src/Schema/SchemaDiff.php

```diff
@@ -162,6 +162,17 @@ protected function _toSql(AbstractPlatform $platform, $saveMode = false)
         }
 
         foreach ($this->changedTables as $tableDiff) {
+            // make sure to drop related orphan foreign keys before dropping indexes that are related to FKs
+            if ($saveMode === true && $platform->supportsForeignKeyConstraints() && $tableDiff->fromTable !== null) {
+                foreach ($this->orphanedForeignKeys as $orphanedForeignKey) {
+                    if ($orphanedForeignKey->getLocalTable()->getName() !== $tableDiff->fromTable->getName()) {
+                        continue;
```

Comment:
If this line is commented out, the test will still pass (at least on MySQL). Is this condition tested?

Reply:
added test for it in https://github.com/doctrine/dbal/pull/4782/commits/9dee9ae2a07ff16bbf9073ee0178648d070f7c01